src/CommandResourceEstimator.cpp

```diff
@@ -0,0 +1,121 @@
+#include "CommandResourceEstimator.hpp"
+#include "CommandRunner.hpp"
+#include "Helpers.hpp"
+
+#include <google/protobuf/text_format.h>
+
+#include <process/defer.hpp>
+#include <process/dispatch.hpp>
+
+namespace criteo {
+namespace mesos {
+
+using namespace process;
+using namespace mesos;
+
+using std::string;
+
+using ::mesos::modules::Module;
+using ::mesos::Resource;
+using ::mesos::Resources;
+using ::mesos::slave::ResourceEstimator;
+using ::mesos::ResourceUsage;
+
+using process::Future;
+using process::Failure;
+using process::terminate;
+
+class CommandResourceEstimatorProcess
+    : public process::Process<CommandResourceEstimatorProcess> {
+ public:
+  CommandResourceEstimatorProcess(
+      const lambda::function<Future<ResourceUsage>()>& _usage,
+      const Option<Command>& oversubscribableCommand, bool isDebugMode);
+
+  virtual process::Future<Resources> oversubscribable();
+  virtual process::Future<Resources> _oversubscribable(
```

Comment:
Even though this identifier is not causing any problems regarding the C++ standard, I would advise against starting an identifier with an underscore altogether.
```
Certain sets of names and function signatures are always reserved to the implementation:

Each name that contains a double underscore __ or begins with an underscore followed by an uppercase letter (2.12) is reserved to the implementation for any use.

Each name that begins with an underscore is reserved to the implementation for use as a name in the global namespace.
```
If you have trouble getting the right pointer to the overloaded function please look at https://en.cppreference.com/w/cpp/language/overloaded_address

Reply:
i have change most of the variable that had an _ in front of the name.